site/src/pages/ManagementSettingsPage/CustomRolesPage/CreateEditRolePageView.tsx

```diff
@@ -0,0 +1,249 @@
+        />
+      </HorizontalForm>
+    </>
+  );
+};
+
+interface ActionCheckboxesProps {
+  permissions: readonly Permission[] | undefined;
+  form: ReturnType<typeof useFormik<Role>> & { values: Role };
+}
+
+const ResourceActionComparator = (
+  p: Permission,
+  resource: string,
+  action: string,
+) =>
+  p.resource_type === resource &&
+  (p.action.toString() === "*" || p.action === action);
+
+const DEFAULT_RESOURCES = [
+  "audit_log",
+  "group",
+  "template",
+  "organization_member",
+  "provisioner_daemon",
+  "workspace",
+];
+
+const resources = new Set(DEFAULT_RESOURCES);
+
+const filteredRBACResourceActions = Object.fromEntries(
+  Object.entries(RBACResourceActions).filter(([resource]) =>
+    resources.has(resource),
+  ),
+);
+
+const ActionCheckboxes: FC<ActionCheckboxesProps> = ({ permissions, form }) => {
+  const [checkedActions, setCheckActions] = useState(permissions);
+  const [showAllResources, setShowAllResources] = useState(false);
+
+  const handleActionCheckChange = async (
+    e: ChangeEvent<HTMLInputElement>,
+    form: ReturnType<typeof useFormik<Role>> & { values: Role },
+  ) => {
+    const { name, checked } = e.currentTarget;
+    const [resource_type, action] = name.split(":");
+
+    const newPermissions = checked
+      ? [
+          ...(checkedActions ?? []),
+          {
+            negate: false,
+            resource_type: resource_type as RBACResource,
+            action: action as RBACAction,
+          },
+        ]
+      : checkedActions?.filter(
+          (p) => p.resource_type !== resource_type || p.action !== action,
+        );
+
+    setCheckActions(newPermissions);
+    await form.setFieldValue("organization_permissions", newPermissions);
+  };
+
+  const resourceActions = showAllResources
+    ? RBACResourceActions
+    : filteredRBACResourceActions;
+
+  return (
+    <>
+      <TableContainer>
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableCell
+                align="right"
+                sx={{ paddingTop: 0.4, paddingBottom: 0.4 }}
+              >
+                <FormControlLabel
+                  sx={{ marginRight: 1 }}
+                  control={
+                    <Checkbox
+                      size="small"
+                      id="show_all_permissions"
+                      name="show_all_permissions"
+                      checked={showAllResources}
+                      onChange={(e) =>
+                        setShowAllResources(e.currentTarget.checked)
+                      }
+                    />
+                  }
+                  label={
+                    <span style={{ fontSize: 12 }}>Show all permissions</span>
+                  }
+                />
+              </TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {Object.entries(resourceActions).map(([resourceKey, value]) => {
+              return (
```

Comment:
```suggestion
            {Object.entries(resourceActions).map(([resourceKey, value]) => (
```

nit: could avoid a level of indention by using single expression/implicit return syntax